### PR TITLE
Add support to Quickstart for testing devices behind an HTTP proxy.

### PR DESCRIPTION
**Pull request by arsing**
This change adds two new CLI parameters `--proxy` and `--upstream-protocol`.
They correspond to `agent.env.https_proxy` and `agent.env.UpstreamProtocol`
in config.yaml.

`--proxy` defaults to the `https_proxy` environment variable on the
`Quickstart` process itself, if it's set.

Also fixed JSON errors in the smoke test deployment templates that caused `jq`
to choke.

### CODE REVIEW

**Comment by arsing** [comment]
This isn't enough. The `EventHubClient` inside `Details` also needs to be configured to use `TransportType.AmqpWebSockets` since the default is `Amqp`

File: smoke/IotEdgeQuickstart/details/PartitionReceiveHandler.cs [hunk 1]
```diff
@@ -26,6 +26,6 @@ public Task ProcessEventsAsync(IEnumerable<EventData> events)
             return Task.CompletedTask;
         }
         public Task ProcessErrorAsync(Exception error) => throw error;
-        public int MaxBatchSize { get; } = 10;
+        public int MaxBatchSize { get; set; }
```
Comment (myagley) [inline smoke/IotEdgeQuickstart/details/PartitionReceiveHandler.cs]: Who uses this setter?

File: smoke/IotEdgeQuickstart/details/Details.cs [hunk 2]
```diff
@@ -216,8 +241,8 @@ protected async Task VerifyDataOnIoTHub(string moduleId)
                 {
                     eventHubReceiver.SetReceiveHandler(new PartitionReceiveHandler(eventData =>
                     {
-                        eventData.Properties.TryGetValue("iothub-connection-device-id", out object devId);
-                        eventData.Properties.TryGetValue("iothub-connection-module-id", out object modId);
+                        eventData.SystemProperties.TryGetValue("iothub-connection-device-id", out object devId);
+                        eventData.SystemProperties.TryGetValue("iothub-connection-module-id", out object modId);
```
[the remark this answers is not in the document]
Comment (arsing) [reply smoke/IotEdgeQuickstart/details/Details.cs]: Yes.

File: smoke/IotEdgeQuickstart/details/PartitionReceiveHandler.cs (hunk 1 above)
Comment (arsing) [reply smoke/IotEdgeQuickstart/details/PartitionReceiveHandler.cs]: The interface requires it.

Comment (arsing) [reply smoke/IotEdgeQuickstart/details/PartitionReceiveHandler.cs]: There doesn't seem to be any reason why [they changed it to require a setter.](https://github.com/Azure/azure-event-hubs-dotnet/commit/7e6b7b9b1d01a2ebc3bfe8513cb43efe10fddf42) The `IPartitionReceiver` impl in the SDK only gets its value.

**Pull request merged**